src/main/java/io/kestra/plugin/nats/Request.java

```diff
@@ -0,0 +1,198 @@
+package io.kestra.plugin.nats;
+
+import io.kestra.core.models.annotations.Example;
+import io.kestra.core.models.annotations.Plugin;
+import io.kestra.core.models.annotations.PluginProperty;
+import io.kestra.core.models.tasks.RunnableTask;
+import io.kestra.core.runners.RunContext;
+import io.nats.client.Connection;
+import io.nats.client.Message;
+import io.nats.client.impl.Headers;
+import io.nats.client.impl.NatsMessage;
+import io.swagger.v3.oas.annotations.media.Schema;
+import lombok.*;
+import lombok.experimental.SuperBuilder;
+
+import jakarta.validation.constraints.NotBlank;
+import jakarta.validation.constraints.NotNull;
+import java.io.InputStream;
+import java.net.URI;
+import java.nio.charset.StandardCharsets;
+import java.time.Duration;
+import java.util.*;
+
+@SuperBuilder
+@ToString
+@EqualsAndHashCode
+@Getter
+@NoArgsConstructor
+@Schema(
+    title = "Send a request to a NATS subject and wait for a reply."
+)
+@Plugin(
+    examples = {
+        @Example(
+            title = "Send a request to the subject and wait for the reply (using username/password authentication).",
+            full = true,
+            code = """
+                id: nats_request_reply
+                namespace: company.team
+
+                tasks:
+                  - id: request
+                    type: io.kestra.plugin.nats.Request
+                    url: nats://localhost:4222
+                    username: nats_user
+                    password: nats_password
+                    subject: "greet.bob"
+                    from:
+                      headers:
+                        someHeaderKey: someHeaderValue
+                      data: "Hello from Kestra!"
+                    requestTimeout: 2000
+                """
+        )
+    }
+)
+public class Request extends NatsConnection implements RunnableTask<Request.Output> {
+    @Schema(
+        title = "Subject to send the request to"
+    )
+    @PluginProperty(dynamic = true)
+    @NotBlank
+    @NotNull
+    private String subject;
+
+    @Schema(
+        title = "Source of message(s) for the request",
+        description = """
+            If this is:
+            - A plain string => entire string is the data
+            - A kestra:// URI => entire file content is read into the data
+            - A list with exactly one item => that item must be a map with optional headers + data
+            - A map => optional 'headers' + 'data' keys
+        """
+    )
+    @NotNull
+    @PluginProperty(dynamic = true)
+    private Object from;
+
+    @Schema(
+        title = "Timeout in milliseconds to wait for a response.",
+        description = "Defaults to 5000 ms."
+    )
+    @Builder.Default
+    private Duration requestTimeout = Duration.ofMillis(5000);
```

Comment:
```suggestion
    private Property<Duration> requestTimeout = Property.of(Duration.ofMillis(5000));
```

You can also add `@NotNull` as it's a required property.